src/components/molecules/community/CommunityBoardTopNavi.tsx

```diff
@@ -1,16 +1,43 @@
 import { useRouter } from 'next/router';
 import IconArrowLeft from '@/components/atoms/IconArrowLeft';
 import { ReactNode } from 'react';
-import IconBookmark from '@/components/atoms/IconBookmark';
 import { colors } from '@/styles/CommunityColors';
+import BookmarkButton from '@/components/atoms/BookmarkButton';
+import { useQuery } from 'react-query';
+import { UrlLangType } from '@/types/common';
+import { getBookmarks } from '@/api/Community';
+import { useBookmarkOnClick } from '@/hooks/useBookmarkOnClick';
 
 export type CommunityBoardTopNaviPropType = {
   boardTitle: string;
+  urlLang: UrlLangType;
+  userId: string;
   rightItem?: ReactNode;
 };
 
-const CommunityBoardTopNavi = ({ boardTitle, rightItem }: CommunityBoardTopNaviPropType) => {
+const CommunityBoardTopNavi = ({
+  boardTitle,
+  urlLang,
+  userId,
+  rightItem,
+}: CommunityBoardTopNaviPropType) => {
   const router = useRouter();
+  const boardIndex = router.query.boardIndex;
+  const { data } = useQuery(['bookmarks', { userId, urlLang }], () =>
+    getBookmarks(userId, urlLang)
+  );
+  const bookmarks = data ?? [];
+  const isBookmarked = Boolean(bookmarks.find((bookmark) => bookmark.BOARD_IDX === boardIndex));
```

Comment:
```suggestion
  const isBookmarked = bookmarks.find((bookmark) => bookmark.BOARD_IDX === boardIndex) ?? false
```

원시타입의 생성자는 브라우저마다 조금씩 다르게 구현되어있기에 의도하지 않은 상황에 놓일 수 있음.